platform/mds/mds/src/main/java/org/motechproject/mds/service/DefaultMotechDataService.java

```diff
@@ -175,6 +179,9 @@ public T retrieve(String primaryKeyName, Object value) {
     @Transactional
     public T update(final T object) {
         validateCredentials(object);
+        if (JDOHelper.getObjectState(object) == ObjectState.TRANSIENT) {
```

Comment:
What if, if in this case, we would try to fetch from the database by ID, then use PropertyUtil to copy properties? If id is not set, or the object not found then throw the exception?